### PR TITLE
New recipe for tin (from empty tin can)

### PR DESCRIPTION
**Pull request by Davi-DeGanne**
Justification is that in absence of finding a metal pot early on (due to low-density/non-existent cities perhaps, or simply low item spawn settings) and no nearby river for clay, the player's only real method to obtain a level 3 cooking tool currently is to set up a forge and forge a steel pot. This is fine, but I wanted to make crafting a copper pot a viable alternate route. In the current build this is entirely dependent on finding a pewter plate or bowl for the tin required, AFAIK the only source of tin in the game. Here, I add a recipe for tin, extracted from empty tin cans via electrolysis, as detailed in [this patent](https://patents.google.com/patent/US511774A/en).

Two things to note: Firstly, I had to introduce a method for electrolysis, since that is an integral part of the process. Standard 1.5V batteries are woefully undervoltage for this application. As vehicle batteries are defined as magazines, I could not use their charges in the recipe directly, so as a workaround I created a new item, "electolysis kit," which accepts vehicle batteries as magazines. Its only use currently, of course, is this recipe. I realize this is a bit of a hack, and if the PR does not get merged due to this I will totally get it. However, if that does end up being the case, I would appreciate some guidance as to where I could go in the C++ code to allow charges from MAGAZINE type items to be used in crafting.

The second note is that I reduced the default charges of tin to 10 from 200, as getting 200 tin per can is absurd. Does this have any effects beyond the amount received from recipes and the amount that appears when it spawns? (Does tin even spawn anywhere? I don't think I've ever seen it)

### CODE REVIEW

**Comment by NotFuji** [comment]
A soldering iron and 50 copper wire is easier to find than a pot?

File: data/json/items/tools.json [hunk 1]
```diff
@@ -8316,5 +8316,19 @@
     "symbol": ",",
     "color": "light_gray",
     "flags": [ "NO_SALVAGE" ]
+  },
+  {
+    "id": "electrolysis_kit",
+    "type": "TOOL",
+    "name": "electrolysis kit",
+    "description": "A set of wiring and electrodes for applying a direct current, usually to a liquid. Useful for crafting. Load with a storage battery or 12V vehicle battery to use.",
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: Please use two spaces after periods.

File: data/json/items/tools.json [hunk 2]
```diff
@@ -8316,5 +8316,19 @@
...
+    "weight": 5,
```
Comment (Night-Pryanik) [inline data/json/items/tools.json]: 5 grams and 0,25 liters for an electrolysis kit? How does it look like, by the way?

File: data/json/recipes/recipe_electronics.json [hunk 3]
```diff
@@ -1983,5 +1983,21 @@
...
+  "type" : "recipe",
+  "result": "electrolysis_kit",
+  "category": "CC_ELECTRONIC",
+  "subcategory": "CSC_ELECTRONIC_TOOLS",
+  "skill_used": "electronics",
+  "using": [ [ "soldering_standard", 10 ] ],
+  "difficulty": 2,
+  "time": 10000,
+  "reversible": true,
+  "autolearn": true,
```
Comment (Night-Pryanik) [inline data/json/recipes/recipe_electronics.json]: Maybe it's better to add this recipe to some book? I feel the electrolysis is rather complicated process to figure it on its own.

File: data/json/items/tools.json (hunk 1 above)
Comment (DaviBones) [reply data/json/items/tools.json]: Ok thanks, will try to remember that in future.

File: data/json/recipes/recipe_electronics.json (hunk 3 above)
Comment (DaviBones) [reply data/json/recipes/recipe_electronics.json]: "Electrolysis kit" in this case is just a fancy name for a couple copper wires hooked up to improvised steel electrodes. Not your typical complex lab setup, any dabbling electrician could figure it out. I think the other recipe might warrant that though, stripping tin by combination of acid + electrolysis is pretty obscure knowledge. Any suggestion for which book?

**Comment by DaviBones** [comment]
@NotFuji In my current game (I've been playing around with city spacing of 8, mixes up the early game a lot) I raided the nearest small town and a couple farmhouses and never found a pot. Unlucky, perhaps. But I did find a power substation and a screwdriver, which is basically all the electronics gear and copper I will ever need. Training electronics up to 2 for a soldering iron was pretty trivial at that point.

I am still open to removing the soldering iron as a requirement though. Do you think that would be better?

File: data/json/recipes/recipe_electronics.json (hunk 3 above)
Comment (Night-Pryanik) [reply data/json/recipes/recipe_electronics.json]: The tin stripping recipe is more of a chemistry, or of an electronics?

Comment (Zireael07) [reply data/json/recipes/recipe_electronics.json]: DIY compendium sounds OK.

File: data/json/recipes/recipe_electronics.json [hunk 4]
```diff
@@ -1983,5 +1983,21 @@
...
+  "qualities" : [ {"id":"CUT","level":1} ],
+  "components": [
+    [ [ "cable", 50 ] ],
+    [ [ "scrap", 2 ] ]
+  ]
```
Comment (NotFuji) [inline data/json/recipes/recipe_electronics.json]: Either the electrolysis kit or the tinplating recipe needs some kind of insulated container (glass).  As-is there's nothing to put the acid bath in.

**Comment by NotFuji** [comment]
Replacing/substituting the copper with jumper cables would make it a little more feasible as a wilderness recipe.  While you might have found a substation in your game, they're extraordinarily rare on average.

On a side note: I'm curious as to how tin helps you get a cooking vessel.  Copper pots still require a hand press & die set, which is arguably a rare item, and requires more items to make than a simple iron pot.  Clay pots can already be made with extremely minimal material.

**Comment by BorkBorkGoesTheCode** [comment]
Why does the copper pot require a hand press to make?

**Comment by DaviBones** [comment]
Thanks for all the feedback guys! I've allowed jumper cables in the electrolysis kit recipe, and made the following changes to the tin recipe:

- Now requires cooking 4 rather than fabrication 2
- No autolearn, found in _chemistry textbook_, _DIY Compendium_, or _Welding and Metallurgy_.
- Requires either a _glass jar_ or a _3L glass jar_, rather than a containing quality of 1.

@NotFuji 
> _Copper pots_ still require a _hand press & die set_
Great point, I had somehow missed that when designing this change. I'm with @BorkBorkGoesTheCode here; I'm baffled as to why you would need either a hand press or a die set to create a crude copper pot. Presumably you would just make a mold from sand, mud, or clay and pour the copper in. What do people think about removing the hand press & die requirement?

**Comment by smolbird** [comment]
I assumed it was for the tin, not the copper. In which case alternatives absolutely should be available.

**Comment by DaviBones** [comment]
@smolbird Sorry, what do you mean? Are you talking about alternatives to the hand press & die set? If so, what alternatives do you suggest? I had just assumed the copper and tin were being alloyed together for a bronze, since the ratio is roughly right for it.

**Comment by BorkBorkGoesTheCode** [comment]
@DaviBones the tin is being used as an acid-resistant liner for the pot interior.

**Comment by DaviBones** [comment]
Oh wow, my bad. Thanks for that. Suppose I ought to read item descriptions more.

Huh.... Turns out the tin-stripping process is reversible in a way. Check it out:

https://hackaday.com/2012/10/14/diy-tin-plating-for-bus-bars/

**Comment by DaviBones** [comment]
Added a new recipe for the copper pot based on the article I linked above.

**Comment by K2705** [comment]
Aren't tin cans made of steel?

**Comment by Zireael07** [comment]
@tyrael93: Tin melts at very very high temperatures, you can't really achieve them on a fire.

**Comment by Soadreqm** [comment]
Pure metallic tin melts at 232°C, which is not very high.
Aren't tin cans made of steel, though?

**Comment by NotFuji** [comment]
Modern ones aren't, they're either made of aluminum or steel lined with plastic.
Even older ones only contain about 2-3 grams of tin per can.

**Comment by DaviBones** [comment]
@tyrael93 The issue would be coating the inside of the pot evenly. With such a small amount of tin, it would be nearly impossible even with modern casting/molding techniques, nevermind primitive ones made of sand/clay/whatever.

@NotFuji Yes, there is a bit of embellishment going on here, but much less embellishment, I would argue, than _the only other source of tin in the game_, pewter tableware -- pewter is an alloy "composed of 85–99% tin, mixed with copper, antimony, bismuth, and sometimes lead," not something you can just "disassemble" into its constituent metals, and certainly not safe for use in cookware as-is (no safer than pure copper at least).

File: data/json/recipes/other/materials.json [hunk 5]
```diff
@@ -27,5 +27,19 @@
     "autolearn": true,
     "using": [ [ "forging_standard", 4 ], [ "steel_tiny", 4 ] ],
     "tools": [ [ [ "crucible", -1 ], [ "crucible_clay", -1 ] ] ]
+  },
+  {
```
Comment (ZhilkinSerg) [inline data/json/recipes/other/materials.json]: Would you format it using http://dev.narc.ro/cataclysm/format.html please?

**Pull request merged**